app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -2485,22 +2485,31 @@ public void onEvents(@NonNull final com.google.android.exoplayer2.Player player,
         Listener.super.onEvents(player, events);
         MediaItemTag.from(player.getCurrentMediaItem()).ifPresent(tag -> {
             if (tag == currentMetadata) {
-                return;
+                return; // we still have the same metadata, no need to do anything
             }
+            final StreamInfo previousInfo = Optional.ofNullable(currentMetadata)
+                    .flatMap(MediaItemTag::getMaybeStreamInfo).orElse(null);
```

Comment:
You can use `getCurrentStreamInfo()` method to do so, unless you want to avoid confusion between the previous info and the current one :)

```suggestion
            final StreamInfo previousInfo = getCurrentStreamInfo().orElse(null);
```

<br>

<details>

<summary>Reminder of getCurrentStreamInfo() method</summary>

```java
    private Optional<StreamInfo> getCurrentStreamInfo() {
        return Optional.ofNullable(currentMetadata).flatMap(MediaItemTag::getMaybeStreamInfo);
    }
```

</details>

Reply:
Even tough to my eye your suggestion looks better, I'd prefer to keep the current code as it shows where the stream info comes from